recipes/stlab/1.7.0_and_above/conanfile.py

```diff
@@ -0,0 +1,218 @@
+        if self.options.task_system == "libdispatch":
+            self._validate_task_system_libdispatch()
+        elif self.options.task_system == "windows":
+            if self.settings.os != "Windows":
+                raise ConanInvalidConfiguration(f"{self.ref} task_system=windows is not supported on {self.settings.os}. Try using task_system=auto")
+
+    def _default_thread_system(self):
+        if self.settings.os == "Macos":
+            return "pthread-apple"
+
+        if self.settings.os == "Linux":
+            return "pthread"
+
+        if self.settings.os == "Windows":
+            return "win32"
+
+        if self.settings.os == "Emscripten":
+            return "pthread-emscripten"
+
+        return "none"
+
+    def _validate_thread_system(self):
+        if self.options.thread_system == "pthread-apple":
+            if self.settings.os != "Macos":
+                raise ConanInvalidConfiguration("{}/{} thread_system=pthread-apple is not supported on {}. Try using thread_system=auto".format(self.name, self.version, self.settings.os))
+        elif self.options.thread_system == "pthread":
+            if self.settings.os != "Linux":
+                raise ConanInvalidConfiguration("{}/{} thread_system=pthread is not supported on {}. Try using thread_system=auto".format(self.name, self.version, self.settings.os))
+        elif self.options.thread_system == "win32":
+            if self.settings.os != "Windows":
+                raise ConanInvalidConfiguration("{}/{} thread_system=win32 is not supported on {}. Try using thread_system=auto".format(self.name, self.version, self.settings.os))
+        elif self.options.thread_system == "pthread-emscripten":
+            if self.settings.os != "Emscripten":
+                raise ConanInvalidConfiguration("{}/{} thread_system=pthread-emscripten is not supported on {}. Try using thread_system=auto".format(self.name, self.version, self.settings.os))
+
+    def _validate_boost_components(self):
+        if self.settings.os != "Macos": return
+        if self.settings.compiler != "apple-clang": return
+        if Version(self.settings.compiler.version) >= "12": return
+        if self.options.with_boost: return
+        #
+        # On Apple we have to force the usage of boost.variant and boost.optional, because Apple's implementation of C++17
+        # is not complete.
+        #
+        raise ConanInvalidConfiguration(f"Apple-Clang versions less than 12 do not correctly support std::optional or std::variant, so we will use boost::optional and boost::variant instead. Try -o {self.name}:with_boost=True.")
+
+    def validate(self):
+        # TODO: No cppstd check until this issue is solved
+        #       https://github.com/conan-io/conan/issues/12210
+        # if self.info.settings.compiler.cppstd:
+        #     check_min_cppstd(self, 17)
+
+        if self.info.settings.compiler == "gcc" and Version(self.info.settings.compiler.version) < "9":
```

Comment:
There is a separation here, when something does not work when being built, like stlab and gcc < 9. When a package can not be installed using a configuration, indeed, it should use validate, but here, it's a tool, which can not be built by gcc, so its correct method is `validate_build` instead, and `self.settings.info` should not be used. 

More info: https://docs.conan.io/en/latest/migrating_to_2.0/recipes.html#the-validate-method